docs/config.md

```diff
@@ -0,0 +1,135 @@
+# Config file options
+
+SOCI has a multitude of variables that can customize SOCI behavior to best suit users' needs. This doc will break down the possible values for each variable, where the variable can be found, and what it can do.
+
+This doc assumes the usage of a TOML file. Per TOML specs, Anything in [brackets] is the required header for the set of variables.
+
+An example configuration file can be found in config/config.toml.
+
+The following format will be used throughout this doc:
+
+## File path
+### StructName [toml_header]
+- `toml_var_name` — StructVarName (type). Brief description. Defaults to some value.
+
+#
+
+# Uncategorized
+
+This set of variables must be at the top of your TOML file due to not belonging to any TOML header.
+
+## config/fs.go
+### FSConfig
+- `http_cache_type` — HTTPCacheType (string). When set to "memory" it will create an HTTP cache in memory, otherwise it will create a cache on the disk. Defaults to empty string. (Unused?)
+- `filesystem_cache_type` — FSCacheType (string). When set to "memory" it will create a filesystem cache in memory, otherwise it will create a cache on the disk. Defaults to empty string. (Unused?)
+- `resolve_result_entry` — ResolveResultEntry (int). Max amount of entries allowed in the cache. Defaults to 0, which sets it to the value of defaultResolveResultEntry (30).
+- `debug` — Debug (bool). Enables debugging for go-fuse in logs. Please note this often emits sensitive data, so this should be false in production. Defaults to false.
+- `allow_no_verification` — AllowNoVerification (bool). Unused variable. Defaults to false.
+- `disable_verification` — DisableVerification (bool). Allows skipping TOC validation, which can give slight performance improvements if files have already been verified elsewhere. Defaults to false.
+- `max_concurrency` — MaxConcurrency (int). Unused variable. Defaults to 0.
+- `no_prometheus` — NoPrometheus (bool). Allows users to not use prometheus metrics. Defaults to false.
+- `mount_timeout_sec` — MountTimeoutSec (int). Amount of time a mount will take to time out if a layer can't be resolved. Defaults to 0, which sets it to the value of defaultMountTimeoutSec (30).
+- `fuse_metrics_emit_wait_duration_sec` — FuseMetricsEmitWaitDurationSec (int). Amount of time the snapshotter will wait before emitting the metrics for a FUSE operation. Defaults to 0, which sets it to the value of defaultMountTimeoutSec (60).
+
+## config/config.go
+### Config
+- `metrics_address` — MetricsAddress (string). Directory to emit Prometheus metrics to. If `no_prometheus=true`, this will have no effect. Defaults to empty string, which means no metrics will be emitted.
```

Comment:
nit: this is the address that the prometheus metrics server listens on. Metrics are polled, not emitted.